src/uct/tcp/sockcm/sockcm_ep.c

```diff
@@ -14,26 +14,201 @@
         } \
     } while (0)
 
+ucs_status_t uct_sockcm_ep_set_sock_id(uct_sockcm_ep_t *ep)
+{
+    ucs_status_t status;
+    struct sockaddr *dest_addr = NULL;
+
+    ep->sock_id_ctx = ucs_malloc(sizeof(*ep->sock_id_ctx), "client sock_id_ctx");
+    if (ep->sock_id_ctx == NULL) {
+        status = UCS_ERR_NO_MEMORY;
+        goto out;
+    }
+    ep->sock_id_ctx->handler_added = 0;
+    dest_addr = (struct sockaddr *) &(ep->remote_addr);
+
+    status = ucs_socket_create(dest_addr->sa_family, SOCK_STREAM,
+                               &ep->sock_id_ctx->sock_id);
+    if (status != UCS_OK) {
+        ucs_debug("unable to create client socket for sockcm");
+        goto out_free;
+    }
+
+    status = UCS_OK;
+    goto out;
+
+out_free:
+    ucs_free(ep->sock_id_ctx);
+out:
+    return status;
+}
+
+static void uct_sockcm_ep_event_handler(int fd, void *arg)
+{
+    ucs_debug("not implemented");
+}
+
 static UCS_CLASS_INIT_FUNC(uct_sockcm_ep_t, const uct_ep_params_t *params)
 {
-    uct_sockcm_iface_t *iface = ucs_derived_of(params->iface,
-                                               uct_sockcm_iface_t);
+    const ucs_sock_addr_t *sockaddr = params->sockaddr;
+    uct_sockcm_iface_t    *iface    = NULL;
+    struct sockaddr *param_sockaddr = NULL;
+    char ip_port_str[UCS_SOCKADDR_STRING_LEN];
+    ucs_status_t status;
+    size_t sockaddr_len;
 
+    iface = ucs_derived_of(params->iface, uct_sockcm_iface_t);
     UCS_CLASS_CALL_SUPER_INIT(uct_base_ep_t, &iface->super);
+
+    if (iface->is_server) {
+        return UCS_ERR_UNSUPPORTED;
+    }
+
+    if (!(params->field_mask & UCT_EP_PARAM_FIELD_SOCKADDR)) {
+        return UCS_ERR_INVALID_PARAM;
+    }
+
+    UCT_SOCKCM_CB_FLAGS_CHECK((params->field_mask &
+			       UCT_EP_PARAM_FIELD_SOCKADDR_CB_FLAGS) ?
+			      params->sockaddr_cb_flags : 0);
+
+    self->pack_cb       = (params->field_mask &
+                           UCT_EP_PARAM_FIELD_SOCKADDR_PACK_CB) ?
+                          params->sockaddr_pack_cb : NULL;
+    self->pack_cb_arg   = (params->field_mask &
+                           UCT_EP_PARAM_FIELD_USER_DATA) ?
+                          params->user_data : NULL;
+    self->pack_cb_flags = (params->field_mask &
+                           UCT_EP_PARAM_FIELD_SOCKADDR_CB_FLAGS) ?
+                          params->sockaddr_cb_flags : 0;
+    pthread_mutex_init(&self->ops_mutex, NULL);
+    ucs_queue_head_init(&self->ops);
+
+    param_sockaddr = (struct sockaddr *) sockaddr->addr;
+    if (UCS_ERR_INVALID_PARAM == ucs_sockaddr_sizeof(param_sockaddr, &sockaddr_len)) {
```

Comment:
I'd compare like `!= UCS_OK`  (some other error codes can be returned if the function gets updated)